server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -397,6 +330,48 @@ public void run() {
             }
         }
 
+        /**
+         * This method evaluates the health info of this node and if there is a change it sends an update request to the health node.
+         *
+         * @return whether the next run has been scheduled.
+         */
+        private boolean sendUpdate(Runnable scheduleNextRun) {
+            ClusterState clusterState = clusterService.state();
+            HealthMetadata healthMetadata = HealthMetadata.getFromClusterState(clusterState);
+            if (healthMetadata == null) {
+                return false;
+            }
+
+            // Create a request builder to which the health checks will add their (changed) health info.
+            var builder = new UpdateHealthInfoCacheAction.Request.Builder().setNodeId(clusterService.localNode().getId());
+            // Perform the health checks and collect all non-null callbacks.
+            var callbacks = healthChecks.stream()
+                .map(healthCheck -> healthCheck.performHealthCheck(builder))
+                .filter(Objects::nonNull)
+                .toList();
+            // If we don't have any callbacks, there were no changes in health, so we don't send an update request.
+            if (callbacks.isEmpty()) {
+                return false;
+            }
```

Comment:
I find this a bit difficult to follow. The variable naming is very helpful and I can understand what it does but I think it could be better.

Can we see if we can separate the concerns a bit more clearly. For example, now the `healthCheck.performHealthCheck(...)` performs a health check and provides passes the result of the health check as a side effect through the builder, but only if it was different than the last observed result. Without putting too much thought I would expect the health check would just return the result of the health check. 

do you want to give it a try to separate concerns a bit more?